aws/resource_aws_opsworks_permission.go

```diff
@@ -125,10 +125,17 @@ func resourceAwsOpsworksPermissionRead(d *schema.ResourceData, meta interface{})
 func resourceAwsOpsworksSetPermission(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*AWSClient).opsworksconn
 
+	/* Level must be supplied as 'nil' if we are not changing it */
+	level_string := d.Get("level").(string)
+	level_aws := aws.String(level_string)
+	if level_string == "" {
+		level_aws = nil
+	}
```

Comment:
We have a more idiomatic way of dealing with this exact problem, take a look at `d.GetOk()`